[Site Isolation] Make storage/indexeddb/modern/blob-cursor.html pass on bot

szewai opened the pull request:
#### b3f95c644b3ba88d2ed59c4023997329c2af80ba
<pre>
[Site Isolation] Make storage/indexeddb/modern/blob-cursor.html pass on bot
<a href="https://bugs.webkit.org/show_bug.cgi?id=291326">https://bugs.webkit.org/show_bug.cgi?id=291326</a>

Reviewed by Ben Nham.

When running this test in cross-origin iframe, exception will be thrown in IDBTransaction creation, because new window
created by `window.open` does not use the same storage partition and its opener and target object store cannot be found.
The error will be handled by global error event handler, where TestRunner::notifyDone() is called and console log about
uncaught exception will be printed.

When site isolation is disabled, result dumping will be immediate (web process =&gt; UI process), so the result does not
have the error console log. When site isolation is enabled, result dumping will be async (subframe process =&gt; UI process
=&gt; main frame process =&gt; UI process), so console log will be sent from subframe process to UI process and will be
included in result. This has led to test failure when running tests with --site-isolation.

To fix this, now we explicitly catch the error in test, so that it will have a more deterministic result regardless of
the implementation.

* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/storage/indexeddb/modern/resources/blob-cursor2.html:
* LayoutTests/storage/indexeddb/modern/resources/blob-svg-image2.html:

Canonical link: <a href="https://commits.webkit.org/293524@main">https://commits.webkit.org/293524@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3a72f301c1401ddfb61ad3461f085b53203cd3be

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99045 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18683 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8970 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104167 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/49629 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18975 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27125 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75394 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32521 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102044 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14423 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89468 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55756 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14216 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7440 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49005 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84149 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7528 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106531 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26142 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-Sequoia-Debug-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84359 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26514 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85668 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83864 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28522 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6221 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/19870 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16126 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26089 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25912 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29232 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27482 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7a50cc60587cce5c3f815850af2d40d440903a33. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/43853

szewai committed:
[Site Isolation] Make storage/indexeddb/modern/blob-cursor.html pass on bot
https://bugs.webkit.org/show_bug.cgi?id=291326

Reviewed by Ben Nham.

When running this test in cross-origin iframe, exception will be thrown in IDBTransaction creation, because new window
created by `window.open` does not use the same storage partition and its opener and target object store cannot be found.
The error will be handled by global error event handler, where TestRunner::notifyDone() is called and console log about
uncaught exception will be printed.

When site isolation is disabled, result dumping will be immediate (web process => UI process), so the result does not
have the error console log. When site isolation is enabled, result dumping will be async (subframe process => UI process
=> main frame process => UI process), so console log will be sent from subframe process to UI process and will be
included in result. This has led to test failure when running tests with --site-isolation.

To fix this, now we explicitly catch the error in test, so that it will have a more deterministic result regardless of
the implementation.

* LayoutTests/platform/mac-site-isolation/TestExpectations:
* LayoutTests/storage/indexeddb/modern/resources/blob-cursor2.html:
* LayoutTests/storage/indexeddb/modern/resources/blob-svg-image2.html:

Canonical link: https://commits.webkit.org/293524@main

webkit-commit-queue commented:
Committed 293524@main (b3f95c644b3b): <https://commits.webkit.org/293524@main>

Reviewed commits have been landed. Closing PR #43853 and removing active labels.
